dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/trackedentityattributevalue/TrackerTrigramIndexingJob.java

```diff
@@ -74,120 +73,91 @@ public void execute(JobConfiguration jobConfiguration, JobProgress progress) {
     TrackerTrigramIndexJobParameters parameters =
         (TrackerTrigramIndexJobParameters) jobConfiguration.getJobParameters();
 
-    log.info(
-        "Starting Trigram Indexing Job. Attributes Provided to Index: {}",
-        parameters.getAttributes());
+    log.info("Starting Trigram Indexing Job.");
     progress.startingProcess("Starting Trigram indexing process");
 
-    // Fetch all indexable attributes only if needed
-    if (!CollectionUtils.isEmpty(parameters.getAttributes()) || !parameters.isSkipIndexDeletion()) {
-      log.debug("Fetching all indexable attributes from db");
-      Set<TrackedEntityAttribute> allIndexableAttributes =
-          trackedEntityAttributeService.getAllTrigramIndexableTrackedEntityAttributes();
-
-      // Trigram index only need if requested in job parameters
-      if (!CollectionUtils.isEmpty(parameters.getAttributes())) {
-        createTrigramIndexesOnIndexableAttributes(progress, parameters, allIndexableAttributes);
-      }
-
-      // Obsolete index deletion
-      if (!parameters.isSkipIndexDeletion()) {
-        removeObsoleteTrigramIndexes(progress, allIndexableAttributes);
-      }
+    log.debug("Fetching all trigram indexed attributes from db");
+    List<Long> indexedAttributes =
+        trackedEntityAttributeTableManager.getAttributeIdsWithTrigramIndex();
+
+    log.debug("Fetching all indexable attributes from db");
+    Set<TrackedEntityAttribute> indexableAttributes =
+        trackedEntityAttributeService.getAllTrigramIndexableTrackedEntityAttributes();
+
+    boolean indexesWereCreated =
```

Comment:
Thoughts on `indexesCreated` instead of `indexesWereCreated` ?

Reply:
I went with `indexesWereCreated` to make it clear it’s a boolean, though `areIndexesCreated` might have been better.
`indexesCreated` makes me think of a collection, which is why I avoided it.
Not a strong preference, fine to change if you think it’s clearer.